.github/workflows/ci.yml

```diff
@@ -21,6 +21,9 @@ jobs:
       - name: Formatting check
         run: cargo fmt --all -- --check
 
+      - name: Run unit tests
+        run: cargo test --all
```

Comment:
Can we move this to a separate builder (e.g., perhaps the windows builder)? I try to keep this one as fast as possible since it's the critical path to deployment.

We generally still block merging PRs on the Windows builder passing, and generally outage is not terrible for perf, so I think that this sort of optimization still makes sense.